src/jtag/drivers/bcm2835gpio.c

```diff
@@ -83,8 +83,12 @@ static inline void bcm2835_delay(void)
 
 static bool is_gpio_config_valid(enum adapter_gpio_config_index idx)
 {
-	/* Only chip 0 is supported, accept unset value (-1) too */
-	return adapter_gpio_config[idx].gpio_num <= 31;
+	/* Only chip 0 is supported, accept unset value (-1) too, allow
+     * any gpio up to including 45 */
+	return adapter_gpio_config[idx].chip_num >= -1
```

Comment:
I wonder if
`adapter_gpio_config[idx].chip_num >= -1 && adapter_gpio_config[idx].chip_num <= 0 && ...`
might be more readable as
`(adapter_gpio_config[idx].chip_num == 0 || adapter_gpio_config[idx].chip_num == -1) && ...`
?

Reply:
agreed